app/components/bookings/booking/booking.component.ts

```diff
@@ -1,20 +1,134 @@
-import {Component, Input} from '@angular/core';
-
-import {Booking} from '../../../models/booking/booking';
-
+import {Component, Input, Output} from '@angular/core';
+import {Booking, BookingSession} from '../../../models/booking/booking';
 import {WorkdaysToHoursPipe} from '../../../pipes/workdays-to-hours.pipe';
+import {IONIC_DIRECTIVES, Alert, NavController, Toast} from "ionic-angular/index";
+import {EditLabel} from "../edit/edit-label.component";
+import Moment = moment.Moment;
+import moment = require("moment/moment");
+import {BookingService} from "../../../services/bookings/booking.service";
+import {Response} from "@angular/http";
+import {CreateBookingComponent} from "../create/create-booking.component";
+import {EventEmitter} from "@angular/common/src/facade/async";
+
 
 @Component({
   selector: 'booking',
+  directives: [IONIC_DIRECTIVES, EditLabel],
   templateUrl: 'build/components/bookings/booking/booking.component.html',
   pipes: [WorkdaysToHoursPipe]
 })
-export class BookingComponent {
-  private _isLiveBooking: boolean = false;
 
+export class BookingComponent{
+
+  // Moment object, but behaves like a String("HH:mm") after set by DateTimePicker via Gui
+  private _pickedEffort :Moment = moment(new Date().getTime());
+  // undefined until user has changed effort
+  private _bookingEffort : number;
+
+  /*
+  * if this.isLife is true, booking is null
+  * otherwise bookingSession will be null
+  * */
+
+  @Input()
+  public withWpName: boolean;
   @Input()
   public booking: Booking;
+  @Input()
+  public bookingSession: BookingSession;
+  @Input()
+  public isLive: boolean;
+  @Output()
+  public deletedBooking = new EventEmitter<Booking>();
+
 
-  constructor() {
+  constructor( private  _bookingService : BookingService, private _nav: NavController) {
   }
+
+  private _changedEffortLabel():void{
+
+    this._bookingEffort = this._momentEffortToWorkdays(this._pickedEffort.toString());
+    this.booking.effort = this._bookingEffort;
+    this._bookingService.update(this.booking).subscribe((returnedBooking: Booking) => {
+      if(returnedBooking)
+      this._showToast("Effort has been updated to " + this._pickedEffort.toString());
+    });
+
+  }
+
```

Comment:
Please remove random paragraphs between methods, to keep code style consistent ;)
Maybe have a look at WebStorms reformat feature!